extensions/wikia/TemplateClassification/TemplateClassification.hooks.php

```diff
@@ -97,26 +95,22 @@ public function onEditPageMakeGlobalVariablesScript( array &$aVars ) {
 	 * @return bool
 	 */
 	public function onEditPageShowEditFormFields( \EditPage $editPage, \OutputPage $out ) {
-		global $wgCityId, $wgEnablePortableInfoboxBuilderExt;
+		global $wgCityId;
 
 		$context = \RequestContext::getMain();
 		$title = $context->getTitle();
 
 		if ( ( new Permissions() )->shouldDisplayEntryPoint( $context->getUser(), $title ) ) {
 			$types = $this->getTemplateTypeForEdit( $editPage->getTitle(), $wgCityId );
 
-			$out->addHTML( \Html::hidden( 'templateClassificationTypeCurrent', $types['current'],
-				['autocomplete' => 'off'] ) );
-			$out->addHTML( \Html::hidden( 'templateClassificationTypeNew', $types['new'],
-				['autocomplete' => 'off'] ) );
+			$out->addHTML( \Html::hidden( 'templateClassificationTypeCurrent', $types[ 'current' ],
+				[ 'autocomplete' => 'off' ] ) );
+			$out->addHTML( \Html::hidden( 'templateClassificationTypeNew', $types[ 'new' ],
+				[ 'autocomplete' => 'off' ] ) );
 
 			// add additional class to body for new templates in order to hide editor while template classification
 			// modal is visible and builder is available
-			if (
-				$wgEnablePortableInfoboxBuilderExt
-				&& $title->getArticleID() === 0
-				&& empty( $types['current'] && $types['new'] )
-			) {
+			if ( $this->shouldHideEditorForInfoboxBuilder( $context, $types ) ) {
 				\OasisController::addBodyClass( self::TC_BODY_CLASS_NAME );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)